src/java/org/apache/cassandra/index/sai/disk/v1/SegmentBuilder.java

```diff
@@ -179,8 +179,11 @@ public boolean isEmpty()
 
         protected long addInternal(ByteBuffer term, int segmentRowId)
         {
-            BytesRefUtil.copyBufferToBytesRef(term, stringBuffer);
-            return ramIndexer.add(stringBuffer.get(), segmentRowId);
+            var encodedTerm = version.onDiskFormat().encodeForOnDiskTrie(term, termComparator);
```

Comment:
I wonder if it is a mistake to put this here instead of before the call to `addInternal`. Thinking...

Reply:
It works because the only types that we encode using the on disk format class are in the trie, and those pass through here. However, it seems surprising to me that an implementation would apply such logic. It seems like it should be up to the SegmentBuilder parent.

Reply:
It might be fine here, but if we wanted to change any other encodings, this might need to move to the method that calls `addInternal`